L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboardME11.cc

```diff
@@ -0,0 +1,853 @@
+                                                   << "]; bx_clct = " << bx_clct << std::endl
+                                                   << "+++ Best CLCT Details: " << clct->bestCLCT[bx_clct] << std::endl
+                                                   << "+++ Second CLCT Details: " << clct->secondCLCT[bx_clct] << std::endl;
+              }
+              if (allLCTs1b(bx_alct,mbx,0).isValid()) {
+                used_clct_mask[bx_clct] += 1;
+                if (match_earliest_clct_only) break;
+              }
+            }
+          }
+        }
+
+        // matching in ME1a
+        if (buildLCTfromCLCTandGEM_ME1a_) {
+          for (int bx_clct = bx_clct_start; bx_clct <= bx_clct_stop; bx_clct++) {
+            if (bx_clct < 0 || bx_clct >= CSCCathodeLCTProcessor::MAX_CLCT_BINS) continue;
+            if (drop_used_clcts && used_clct_mask_1a[bx_clct]) continue;
+            if (clct1a->bestCLCT[bx_clct].isValid()){
+              const int quality(clct1a->bestCLCT[bx_clct].getQuality());
+              // only use high-Q stubs for the time being
+              if (quality < 4) continue;
+              int mbx = bx_clct-bx_clct_start;
+	      CSCGEMMotherboard::correlateLCTsGEM(clct1a->bestCLCT[bx_clct], clct1a->secondCLCT[bx_clct], coPads,
+						  allLCTs1a(bx_alct,mbx,0), allLCTs1a(bx_alct,mbx,1), ME1A);
+              if (debug_matching) {
+                //	    if (infoV > 1) LogTrace("CSCMotherboard")
+                LogTrace("CSCGEMCMotherboardME11") << "Successful GEM-CLCT match in ME1a: bx_alct = " << bx_alct
+                                                   << "; match window: [" << bx_clct_start << "; " << bx_clct_stop
+                                                   << "]; bx_clct = " << bx_clct << std::endl
+                                                   << "+++ Best CLCT Details: " << clct1a->bestCLCT[bx_clct] << std::endl
+                                                   << "+++ Second CLCT Details: " << clct1a->secondCLCT[bx_clct] << std::endl;
+              }
+              if (allLCTs1a(bx_alct,mbx,0).isValid()){
+                used_clct_mask_1a[bx_clct] += 1;
+                if (match_earliest_clct_only) break;
+              }
+            }
+          }
+        }
+      }
+    }
+  } // end of ALCT-centric matching
+
+  if (debug_matching){
+    LogTrace("CSCGEMCMotherboardME11") << "========================================================================" << std::endl
+                                       << "Counting the LCTs" << std::endl
+                                       << "========================================================================" << std::endl;
+  }
+
+  // reduction of nLCTs per each BX
+  for (int bx = 0; bx < MAX_LCT_BINS; bx++)
+  {
+    // counting
+    unsigned int n1a=0, n1b=0;
+    for (unsigned int mbx = 0; mbx < match_trig_window_size; mbx++)
+      for (int i=0;i<2;i++)
```

Comment:
"2" appears quite often - could it be redefined as a constant (or better derived from LCTContainer itself?

Reply:
This has been updated. Various instances of "2" were replaced by MAX_LCT_PER_CSC